CMakeLists.txt

```diff
@@ -627,21 +627,14 @@ if(WITH_INFLATE_ALLOW_INVALID_DIST)
 endif()
 
 #
-# Macro to add either the given intrinsics option to the global compiler options,
+# Macro to add the given intrinsics option to the specified files,
 # or ${NATIVEFLAG} (-march=native) if that is appropriate and possible.
-# An alternative version of this macro would take a file argument, and set ${flag}
-# only for that file as opposed to ${NATIVEFLAG} globally, to limit side-effect of
-# using ${flag} globally.
 #
 macro(add_intrinsics_option flag)
     if(WITH_NATIVE_INSTRUCTIONS AND NATIVEFLAG)
-        if(NOT "${CMAKE_C_FLAGS} " MATCHES ".*${NATIVEFLAG} .*")
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${NATIVEFLAG}")
-        endif()
+        set_property(SOURCE ${ARGN} PROPERTY COMPILE_FLAGS ${NATIVEFLAG})
```

Comment:
Do you want APPEND here?

Reply:
APPEND might result in duplicate or conflicting flags. Most compile flags are set on directory level, not source file level, so this line overrides directory level flags.

Reply:
I'm unsure.

Reply:
I had the same concern as @dankegel while reading this line. As it is today, I don't see a problem since `add_intrinsics_option` is only called once for each group of source files, so nothing is being overwritten. But the name of the macro may convey the wrong idea. At least for me, without reading the macro implementation, I would think it would just append to the list of flags. Maybe rename it to `set_intrinsics_option`?

Reply:
+1 on renaming to match function.

Reply:
Good idea! Fixed.

Reply:
We can also possibly remove the macro all together when we fix the issue with the native flag in another PR..